src/topology/metric_space/antilipschitz.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2020 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import topology.metric_space.lipschitz
+
+/-!
+# Antilipschitz functions
+
+We say that a map `f : α → β` between two (extended) metric spaces is
+`antilipschitz_with K`, `K ≥ 0`, if for all `x, y` we have `K * edist x y ≤ edist (f x) (f y)`.
+For a metric space, the latter inequality is equivalent to `K * dist x y ≤ dist (f x) (f y)`.
+
+## Implementation notes
+
+The parameter `K` has type `nnreal`. This way we avoid conjuction in the definition and have
+coercions both to `ℝ` and `ennreal`. We do not require `0 < K` in the definition, mostly because
+we do not have a `posreal` type.
+-/
+
+variables {α : Type*} {β : Type*} {γ : Type*}
+
+open_locale nnreal
+
+/-- We say that `f : α → β` is `antilipschitz_with K` if for any two points `x`, `y` we have
+`K * edist x y ≤ edist (f x) (f y)`. -/
+def antilipschitz_with [emetric_space α] [emetric_space β] (K : ℝ≥0) (f : α → β) :=
+∀ x y, ↑K * edist x y ≤ edist (f x) (f y)
+
+lemma antilipschitz_with_iff_mul_dist_le [metric_space α] [metric_space β] {K : ℝ≥0} {f : α → β} :
+  antilipschitz_with K f ↔ ∀ x y, ↑K * dist x y ≤ dist (f x) (f y) :=
+by { simp only [antilipschitz_with, edist_nndist, dist_nndist], norm_cast }
+
+alias antilipschitz_with_iff_mul_dist_le ↔ antilipschitz_with.mul_dist_le
+  antilipschitz_with.of_mul_dist_le
+
+namespace antilipschitz_with
+
+variables [emetric_space α] [emetric_space β] [emetric_space γ] {K : ℝ≥0} {f : α → β}
+
+protected lemma injective (hf : antilipschitz_with K f) (hK : 0 < K) :
+  function.injective f :=
+begin
+  intros x y h,
+  have := hf x y,
+  rw [h, edist_self, le_zero_iff_eq, ennreal.mul_eq_zero, ennreal.coe_eq_zero, edist_eq_zero] at this,
+  exact this.elim (λ hK', absurd hK' (ne_of_gt hK)) id
+end
+
+lemma edist_le (hf : antilipschitz_with K f) (hK : 0 < K) (x y) :
```

Comment:
For `hf : lipschitz_with K f`, `hf.edist_le` is the defining property while `edist_ge` is a version with `K` moved to another side of the inequality. Here `edist_le` is a version with `K` moved to the right. Suggestions of better names are welcome.

Reply:
You could add a `mul` on the side where there is the constant, and use `edist` to refer to the edistance of the images. So, for a Lipschitz function, the inequality `edist (f x) (f y) <= K * edist x y` would be `edist_le_mul`, while your inequality here would be `le_mul_edist`, and the defining inequality for antilipschitz functions would be `mul_le_edist`, and so on.